lib/matplotlib/tests/test_sphinxext.py

```diff
@@ -122,6 +122,50 @@ def test_plot_html_show_source_link(tmpdir):
     assert "index-1.py" not in [p.name for p in html_dir2.iterdir()]
 
 
+@pytest.mark.parametrize('plot_html_show_source_link', [0, 1])
+def test_show_source_link_true(tmpdir, plot_html_show_source_link):
+    # Test that a source link is generated if :show-source-link: is true,
+    # whether or not plot_html_show_source_link is true.
+    source_dir = Path(tmpdir) / 'src'
+    source_dir.mkdir()
```

Comment:
I think you can get rid of making a `source_dir` subdirectory and instead work in the `tmp_path` immedtately, which makes these two lines obsolete.

Reply:
Done.